pcfuncs/funclib/tiles.py

```diff
@@ -152,8 +154,82 @@ async def create(
 
 
 class GDALTileSet(TileSet[GDALRaster]):
+    async def _get_tile(self, url: str) -> Union[RIOImage, None]:
+        async def _f() -> RIOImage:
+            async with aiohttp.ClientSession() as session:
+                async with self._async_limit:
+                    # We set Accept-Encoding to make sure the response is compressed
+                    async with session.get(
+                        url, headers={"Accept-Encoding": "gzip"}
+                    ) as resp:
+                        if resp.status == 200:
+                            with MemoryFile(io.BytesIO(await resp.read())) as mem:
+                                with mem.open() as src:
+                                    return RIOImage.from_rio(src)
```

Comment:
we assume the output from the tile is a raster that can be read by rasterio (MemoryFile)